librz/core/golang.c

```diff
@@ -826,6 +826,421 @@ static ut32 golang_recover_string_arm32(GoStrRecover *ctx) {
 	return nlen;
 }
 
+// Possible mips32 signatures
+// -------
+// lui   v0, high_string_offset
+// addiu v0, v0, low_string_offset
+// sw    v0, 0x08(at)
+// addiu v0, zero, string_size
```

Comment:
is this the right thing to do for each arch? Isn't ESIL/emulation/RzIL or whatever already a generic purpose solution to get these kind of things?

Reply:
no, each arch has different ops. this could be emulated but once RzIL is ported.

Reply:
It doesn't sound like a generic enough approach IMHO. You could say the same of many other strings in many different compilers... I feel it is too compiler specific and we are just going to find problems going forward. IMHO we should just aim for a generic solution.

Reply:
actually it is not compiler optimized. do not ask me how.